src/app.tsx

```diff
@@ -12,16 +12,17 @@ import { ContractsProvider } from "./contexts/contracts/contracts-provider";
 import { TemplateSidebar } from "./components/page-templates/template-sidebar";
 import { EthWallet } from "./components/eth-wallet";
 import { VegaWallet } from "./components/vega-wallet";
+import { DataManager } from "./data-manager";
 
 function App() {
   return (
     <GraphQlProvider>
       <Router>
         <Web3Provider>
-          <AppStateProvider>
-            <ContractsProvider>
+          <ContractsProvider>
+            <AppStateProvider>
               <AppLoader>
-                <>
+                <DataManager>
```

Comment:
Optional: can we name this any better? Maybe TokenManager/ContractData or something a bit more specific than just data?

Reply:
Renamed to BalanceManager. Also moved to components folder